src/DirectAssetEnqueuer.php

```diff
@@ -4,34 +4,191 @@
 
 class DirectAssetEnqueuer extends AbstractAssetEnqueuer
 {
-    /** @var AssetManager */
-    protected $assetsManager;
+    private $publicPath;
+    /**
+     * @var WordpressAssetGateway
+     */
+    private $wordpressAssetGateway;
+
+    /**
+     * @param AssetManager $assetManager
+     * @param string $publicPath Path to asset location
+     * @param WordpressAssetGateway|null $wordpressAssetGateway
+     */
+    public function __construct(AssetManager $assetManager, string $publicPath, WordpressAssetGateway $wordpressAssetGateway = null)
+    {
+        parent::__construct($assetManager);
+
+        $this->assetManager->callServices();
+        $this->publicPath = $publicPath;
+        if ($wordpressAssetGateway === null) {
+            $this->wordpressAssetGateway = new WordpressAssetGateway();
+        } else {
+            $this->wordpressAssetGateway = $wordpressAssetGateway;
+        }
+        $this->register();
+    }
+
+    /**
+     * Make js/css available to be able to queue them later
+     */
+    private function register(): void
+    {
+        $this->registerStyles();
+        $this->registerScripts();
+    }
+
+    private function registerStyles(): void
+    {
+        $cssToRegister = [];
+
+        foreach ($this->assetManager->getDependencies('css') as $dep) {
+            /** @var Asset $dep */
+            $cssToRegister[$dep->handle] = [
+                'handle' => $dep->handle,
+                'src' => $dep->src,
+                'deps' => $dep->deps,
+                'ver' => $dep->ver,
+            ];
+        }
+
+        $cssToRegister = $this->wordpressAssetGateway->applyFilters('wwp.enqueuer.register.cssAssets', $cssToRegister);
```

Comment:
Vu que le filtre a le même nom partout (ce qui me semble juste), il faudrait qu'on passe l'enqueur en param si jamais on avait envie de faire des traitements différents en fonction du contexte d'enqueue.

```suggestion
        $cssToRegister = $this->wordpressAssetGateway->applyFilters('wwp.enqueuer.register.cssAssets', $cssToRegister,$this);
```